### ISSUE TITLE
need to plan out some of the main features

### ISSUE THREAD

**Issue by getify** [getify/youperiod.app#5]
Here's a few ideas for initial features:

1. we're going to need a way to easily see ranges of dates in the future for projected/expected periods, as well as the ranges in the past for actual periods (and compared to what was projected).

2. There will also be a sort of "daily check-in" during a period (or the days leading up to it) for self-reporting any symptoms or status of the menstrual flow, etc.

But especially for those who have seen/used period-tracking apps already, please speak up and help us plan out what features are needed, and details about how they need to work to be useful and as easy as possible UX wise.

**Comment by aditiranka27** [comment]
As a user of period tracking apps from quite a long time, one of the feature I would suggest would be logging in symptoms (like headache, cramps, nausea, body ache, mood swings etc.)

**Comment by getify** [comment]
@Kath-1 definitely agree, been thinking about this for sure. I. have ideas for how to do so. securely (by storing/retrieving the encypted dataset from the serve).

Even though it would be encrypted (and https), introducing any server besides static file serving, does somewhat detract from the obvious trust boundary this app provides.

Also, contemplating if device. sync could instead be peer-to-peer using webrtc, which reduces the security concern if the data never even touches our server. 

So it's a delicate balance between convenience and safety.

It will have to be optional and off by default. And we'll have to be very careful how we explain the optional capabilities if we offer them.

But I definitely have this in mind for the roadmap. Thanks for bringing it up!

**Comment by crs1138** [comment]
Many women make a note of their vaginal temperature during their cycle as part of their conception/contraception planning. Perhaps it could be a feature.

**Comment by yepyeplabs** [comment]
a webhook for receiving tracking information. i use this to make things more accessible by voice "hey voice assistant, do this" without the app itself needing to be accessible in that way. data entry sucks when one struggles with mobility

**Comment by getify** [comment]
@yepyeplabs I don't understand what you mean by "webhook" here. Can you elaborate?

**Comment by yepyeplabs** [comment]
yes, it's a no-code term used to describe the ability to receive an HTTP request.

**Comment by getify** [comment]
@yepyeplabs perhaps we might take the discussion to another thread, but... is that an http request that the **CLIENT** somehow receives? Or is it that your server handles it and forwards something to the client?

**Comment by Gynko** [comment]
I had a conversation with my girlfriend yesterday about features.
She mentioned:
- Having a calendar that shows when the periods started and how long they lasted. She mentioned she was especially focused on when the first day of periods was
- Annotate symptoms encountered during a cycle
- Stats: Time elapsed since the last periods (more precisely since the first day of the last periods, not the last)
- Stats: average gap between periods, average periods range time
- Having some health facts about periods: general knowledge or possible symptoms 
- Features related to ovulation?
- For young people that have not had their first periods or just got them: some advices about what to do, how to speak with caretakers

**Comment by artmarydotir** [comment]
I don't know if this has been discussed or not.

But there is two main range
1: Period Length (average:7days)
2: Cycle length (average: 28days)

**Which is different for everyone.**
And we have to get this.
like we need a config section and first of all we should get these information. As long as we get close to the next period, let us know a few days earlier. ( send notification )
Or if we have exceeded its length, let us know that the period has been delayed. ( send notification )

***
There is another important possibility that can be **calculated**.
On days 9 to 19 of the cycle, the probability of pregnancy is higher than on other days.
ّI mean there are days in every month when the chances of getting pregnant are higher or lower.( called Fertility Window )
If we mark the calendar of those days in a way that is distinct, we will provide users with very good information.

***
Another feature that can be useful
Put a pregnancy option in the settings.
If a person becomes pregnant, she can activate the option and no longer receive period notifications.

*** 
Last one is:
having multilingual functionality.

**Comment by pradeeps4ini** [comment]
If there is going to be multilingual functionality in the app, we should use logical properties and values in the CSS. They will help to design layout according to the flow on the text.

**Comment by artnerdnet** [comment]
I just wanted to add a +1 on the notifications! 
I really like the fitbit tracker because apart from notifying me when my period is coming, it shows me how many days left when I open the app. If I don't log it, a day after it was supposed to start, it will ask me if I got my period. 
It would be also be very helpful to have a notification if the person haven't logged the end of the period, for example when it has exceeded their usual length. 
I don't know if you have discussed this yet but I think it would be very nice to have an onboarding screen when you just download the app so you can fill in some basic info to customize the experience (like how long your periods usually last and such).
Another idea: All my friends use an alarm on their phones to remind them to take their contraceptives, maybe it would be a nice to have in the future, or just the option of being able to log when you're taking something since it helps to be more mindful when you see changes in your period. 

Here are some other features that I've seen and I really liked so far from other apps:

### Symptoms logging
Some of the symptoms that I found useful to log are: 
- Menstrual pain
- Headaches
- Acne
- Bloated/water retention
- Back pain
- Mood changes
- Upset stomach

### Flow and discharge
- Intensity of the menstrual flow (spots, light, medium, heavy)

- Discharge/cervical mucus throughout the month (dry, sticky, creamy, clear, egg white) -> https://my.clevelandclinic.org/health/body/21957-cervical-mucus

**Comment by rpalaciosg** [comment]
I want to add something to the comments, my wife mentions that an option could be added that allows to say if during the Cyclo an emergency anticopseptive was taken and to be able to control (calculate in the future) in those cases, usually not always and not to everyone it usually brings hormonal changes and starts an irregular Cyclo.
